Note that the prioritizetransaction dummy value is deprecated, and has no meaning

TheBlueMatt opened the pull request:
This is an alternative version of #10252, explaining that the first value is "dummy" instead of adding more references to priority now that we've removed it.

sipa commented on src/rpc/mining.cpp:
```diff
@@ -257,26 +257,31 @@ UniValue getmininginfo(const JSONRPCRequest& request)
             "\nArguments:\n"
             "1. \"txid\"       (string, required) The transaction id.\n"
-            "2. fee_delta      (numeric, required) The fee value (in satoshis) to add (or subtract, if negative).\n"
+            "2. dummy          (numeric, optional) API-Compaibility for previous API. Must be zero or null.\n"
```
Typo: Compaibility

luke-jr commented on src/rpc/client.cpp:
```diff
@@ -112,7 +112,8 @@ static const CRPCConvertParam vRPCConvertParams[] =
     { "estimaterawfee", 1, "threshold" },
     { "estimaterawfee", 2, "horizon" },
-    { "prioritisetransaction", 1, "fee_delta" },
+    { "prioritisetransaction", 1, "dummy" },
```
This breaks the API compatibility...

TheBlueMatt replied:
I dont think we care for position-independant arguments? They're still super new and I would be highly surprised if anyone is prioritizing by priority in 0.14, where its highly unsupported and disabled by default?

jnewbery commented on src/rpc/mining.cpp:
```diff
@@ -257,26 +257,31 @@ UniValue getmininginfo(const JSONRPCRequest& request)
+    CAmount nAmount = request.params[2].get_int64();
+
+    if (!(request.params[1].isNull() || request.params[1].get_real() == 0)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Priority is no longer supported, adjustment thereof must be zero.");
```
`adjustment thereof` sounds a little tortured to me. How about `The second argument must be set to 0.0`

(simpler English and specifies precisely what the argument should be set to)

jnewbery commented on src/rpc/mining.cpp:
```diff
@@ -257,26 +257,31 @@ UniValue getmininginfo(const JSONRPCRequest& request)
             "\nArguments:\n"
             "1. \"txid\"       (string, required) The transaction id.\n"
-            "2. fee_delta      (numeric, required) The fee value (in satoshis) to add (or subtract, if negative).\n"
+            "2. dummy          (numeric, optional) API-Compatibility for previous API. Must be zero or null.\n"
```
Suggest you add a note that the second (dummy) argument is deprecated and may be removed in a future version. Clients should change to calling this RPC with named `fee_delta` argument if possible for forward compatibility.

jnewbery reviewed (commented):
tested ACK 99445847d43da86724aa49ef193f57c63a97f17b . A couple of documentation suggestions inline.

MarcoFalke commented:
Needs rebase if still relevant

TheBlueMatt committed:
Remove references to priority that snuck back in in 870824e9.

The "priority" field should be appropriately marked as a "dummy"
value and noted that it is deprecated and will likely be removed.

TheBlueMatt commented:
Rebased.

luke-jr commented on src/rpc/mining.cpp:
```diff
@@ -280,7 +281,7 @@ UniValue prioritisetransaction(const JSONRPCRequest& request)
 
     if (!(request.params[1].isNull() || request.params[1].get_real() == 0)) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Priority is not supported, and adjustment thereof must be zero.");
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Priority is no longer supported, first argument to prioritisetransaction must be 0.");
```
Using named arguments, "first argument" may be confusing. Also suggest the recommendation should be null, not 0.

TheBlueMatt replied:
Fixed.

luke-jr commented on src/rpc/mining.cpp:
```diff
@@ -964,7 +965,7 @@ static const CRPCCommand commands[] =
     { "mining",             "getnetworkhashps",       &getnetworkhashps,       true,  {"nblocks","height"} },
     { "mining",             "getmininginfo",          &getmininginfo,          true,  {} },
-    { "mining",             "prioritisetransaction",  &prioritisetransaction,  true,  {"txid","priority_delta","fee_delta"} },
+    { "mining",             "prioritisetransaction",  &prioritisetransaction,  true,  {"txid","dummy","fee_delta"} },
```
This breaks the API compatibility. IMO leave it alone here, and just update the help.

MarcoFalke replied:
Compatibility is already broken, as we require `priority_delta==0`

luke-jr replied:
Hmm, good point.

ryanofsky reviewed (commented):
utACK 40796e1a9d01758ba61d0d68fd05574a4272e4eb.

Maybe rename PR to something like "Improve documentation for prioritisetransaction" since this isn't really about restoring API compatibility anymore.

Also, if you wanted to update the release notes here to mention the priority_delta->dummy rename, and requirement to pass 0 or null, that would be good.

TheBlueMatt commented:
Mentioned the release notes at #9889, updated PR title.

sipa commented:
utACK 40796e1a9d01758ba61d0d68fd05574a4272e4eb

laanwj commented:
utACK 40796e1
